code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -22,111 +24,155 @@
+
+    private final int shooterRPM; //
+    private SpeedController m_wheels;
     private SpeedController m_kicker;
+    private SpeedControllerGroup m_feeder;
+    private boolean m_feedBackward = false;
 
     @Inject
     public ShooterSubsystem(ILogger logger, final RobotConfig config) {
-
-
         super(logger);
-        _config = config;
-        m_leftShooter = new CANSparkMax(_config.Shooter.leftShooterPort, MotorType.kBrushless);
-        m_rightShooter = new CANSparkMax(_config.Shooter.rightShooterPort, MotorType.kBrushless);
+        m_leftShooter = new CANSparkMax(config.Shooter.leftShooterPort, MotorType.kBrushless);
+        m_rightShooter = new CANSparkMax(config.Shooter.rightShooterPort, MotorType.kBrushless);
         m_leftShooter.setInverted(true);
-        m_rightShooter.setSmartCurrentLimit(35, 60, 150);
-        m_leftShooter.setSmartCurrentLimit(35, 60, 150);
+        m_rightShooter.setSmartCurrentLimit(config.Shooter.stallLimit, config.Shooter.freeLimit,
+                config.Shooter.limitRPM);
+        m_leftShooter.setSmartCurrentLimit(config.Shooter.stallLimit, config.Shooter.freeLimit,
+                config.Shooter.limitRPM);
         m_leftEncoder = m_leftShooter.getEncoder();
-        m_rightEncoder = m_rightShooter.getEncoder();
         m_shooter = new SpeedControllerGroup(m_leftShooter, m_rightShooter);
-        m_shooterPID = new PIDController(0.0003, 0.0004, 0);
-        m_kicker = new WPI_VictorSPX(_config.Shooter.kickerPort);
-        m_shooterPID.setTolerance(10);
-        m_shooterPID.reset();
+        m_shooterPID = new PIDController(config.Shooter.pidP, config.Shooter.pidI, config.Shooter.pidD);
+        m_kicker = new WPI_VictorSPX(config.Shooter.feederPort);
+        m_wheels.setInverted(true);
+        m_feeder = new SpeedControllerGroup(m_kicker, m_wheels);
 
-        // TODO Auto-generated constructor stub
+        m_shooterPID.setTolerance(config.Shooter.pidTolerance);
+        m_shooterPID.reset();
+        pidOnTargetTicks = config.Shooter.pidOnTargetTicks;
+        shooterRPM = config.Shooter.shooterRPM;
     }
 
     /*
-     * @param targetVelocity the rpm of shooter
      */
     @Override
-    public void setSpeed(double targetVelocity) {
-        m_shooterPID.setSetpoint(targetVelocity);
-        m_targetVelocity = targetVelocity;
+    public void turnOnShooter() {
+        if (m_shooterPID.getSetpoint() != 0) {
+            return;
+        }
         m_shooterPID.reset();
+        int setpoint = Preferences.getInstance().getInt("shooterTargetRPM", shooterRPM);
+        m_shooterPID.setSetpoint(setpoint);
+        SmartDashboard.putNumber("ShooterTargetRPM", setpoint);
     }
 
+    /**
+     * basically, this makes sure one ball is yeeted, if feedbackward is false. if
+     * feedbackward is true, it just drives the feeder in the opposite direction,
+     * probably becasue mech didn't fix their problems, and the balls got jammed
+     */
```

Comment:
what is this corresponding to?

Reply:
that boolean shouldnt be there why is it there??

Reply:
it's fine for now.  if it works, just leave it